kernel/src/table.rs

```diff
@@ -76,6 +77,11 @@ impl Table {
     pub fn snapshot(&self, engine: &dyn Engine, version: Option<Version>) -> DeltaResult<Snapshot> {
         Snapshot::try_new(self.location.clone(), engine, version)
     }
+
+    /// Create a new write transaction for this table.
+    pub fn new_transaction(&self, engine: &dyn Engine) -> DeltaResult<Transaction> {
```

Comment:
In java kernel, we found the `TransactionBuilder` to be useful. Would it be worth adding a simple builder now? (Do you plan on having one in the future?) I wonder if too much code begins to depend on `new_transaction` returning a `Transaction` it may be a pain for us to refactor it to a builder later.

Also, if a Builder is an anti pattern in rust or something, please let me know.

Reply:
definitely not an anti-pattern but for now just doing the simpler non-builder approach. in the future, this API would just likely return a `TransactionBuilder` and be pretty trivial to migrate to